ledger_secure_sdk_sys/build.rs

```diff
@@ -71,48 +71,217 @@ const CCID_FILES: [&str; 9] = [
+
+    for line in BufReader::new(target_file).lines().flatten() {
+        if target_id.is_none() && line.contains("TARGET_ID") {
+            target_id = Some(
+                line.split_whitespace()
+                    .nth(2)
+                    .ok_or("err")
+                    .map_err(|_| SDKBuildError::MissingTargetId)?
+                    .to_string(),
+            );
+        } else if target_name.is_none() && line.contains("TARGET_") {
+            target_name = Some(
+                line.split_whitespace()
+                    .nth(1)
+                    .ok_or("err")
+                    .map_err(|_| SDKBuildError::MissingTargetName)?
+                    .to_string(),
+            );
+        }
+
+        if target_id.is_some() && target_name.is_some() {
+            // Both tokens found, break out of the loop
+            break;
+        }
+    }
+
+    let target_id = target_id.ok_or(SDKBuildError::MissingTargetId)?;
+    let target_name = target_name.ok_or(SDKBuildError::MissingTargetName)?;
+    Ok((target_id, target_name))
+}
+
 /// Fetch the appropriate C SDK to build
-fn clone_sdk(device: &Device) -> (PathBuf, u32) {
+fn clone_sdk(device: &Device) -> Result<SDKInfo, SDKBuildError> {
+    let mut sdk_info = SDKInfo::new();
     let (repo_url, sdk_branch, api_level) = match device {
-        Device::NanoS => ("https://github.com/LedgerHQ/nanos-secure-sdk", "master", 0),
+        Device::NanoS => (
+            Path::new("https://github.com/LedgerHQ/nanos-secure-sdk"),
+            "master",
+            0,
+        ),
         Device::NanoX => (
-            "https://github.com/LedgerHQ/ledger-secure-sdk",
+            Path::new("https://github.com/LedgerHQ/ledger-secure-sdk"),
             "API_LEVEL_5",
             5,
         ),
         Device::NanoSPlus => (
-            "https://github.com/LedgerHQ/ledger-secure-sdk",
+            Path::new("https://github.com/LedgerHQ/ledger-secure-sdk"),
             "API_LEVEL_1",
             1,
         ),
     };
 
     let out_dir = env::var("OUT_DIR").unwrap();
-    let bolos_sdk = Path::new(out_dir.as_str()).join("ledger-secure-sdk");
-    if !bolos_sdk.exists() {
+    sdk_info.bolos_sdk = Path::new(out_dir.as_str()).join("ledger-secure-sdk");
+    if !sdk_info.bolos_sdk.exists() {
         Command::new("git")
             .arg("clone")
-            .arg(repo_url)
+            .arg(repo_url.to_str().unwrap())
             .arg("-b")
             .arg(sdk_branch)
-            .arg(bolos_sdk.as_path())
+            .arg(sdk_info.bolos_sdk.as_path())
             .output()
             .ok();
     }
-    (bolos_sdk, api_level)
+    sdk_info.api_level = api_level;
+    (sdk_info.target_id, sdk_info.target_name) =
+        retrieve_target_file_infos(device, &sdk_info.bolos_sdk)?;
+    (sdk_info.c_sdk_hash, sdk_info.c_sdk_version) = retrieve_sdk_git_info(&sdk_info.bolos_sdk);
+    sdk_info.c_sdk_name = repo_url.file_name().unwrap().to_str().unwrap().to_string();
```

Comment:
You could `retrieve_sdk_info` here I guess?
And I even think that we could do it directly in `pub fn bolos_sdk` either with the path to the already present SDK, or with the path of the cloned SDK (path that would be the only return of `clone_sdk`

Reply:
Done.